code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -60,34 +61,57 @@
 	var/datum/reagents/metabolism/active_metab = location
 	var/removed = metabolism
 
+	var/ingest_rem_mult = 1
+	var/ingest_abs_mult = 1
+
 	if(!mrate_static == TRUE)
 		// Modifiers
 		for(var/datum/modifier/mod in M.modifiers)
 			if(!isnull(mod.metabolism_percent))
 				removed *= mod.metabolism_percent
+				ingest_rem_mult *= mod.metabolism_percent
 		// Species
 		removed *= M.species.metabolic_rate
+		ingest_rem_mult *= M.species.metabolic_rate
 		// Metabolism
 		removed *= active_metab.metabolism_speed
+		ingest_rem_mult *= active_metab.metabolism_speed
 
 		if(ishuman(M))
 			var/mob/living/carbon/human/H = M
-			if(H.species.has_organ[O_HEART])
+			if(H.species.has_organ[O_HEART] && (active_metab.metabolism_class == CHEM_BLOOD))
 				var/obj/item/organ/internal/heart/Pump = H.internal_organs_by_name[O_HEART]
 				if(!Pump)
 					removed *= 0.1
 				else if(Pump.standard_pulse_level == PULSE_NONE)	// No pulse normally means chemicals process a little bit slower than normal.
 					removed *= 0.8
 				else	// Otherwise, chemicals process as per percentage of your current pulse, or, if you have no pulse but are alive, by a miniscule amount.
 					removed *= max(0.1, H.pulse / Pump.standard_pulse_level)
+
+			if(H.species.has_organ[O_STOMACH] && (active_metab.metabolism_class == CHEM_INGEST))
+				var/obj/item/organ/internal/stomach/Chamber = H.internal_organs_by_name[O_STOMACH]
+				if(Chamber)
+					ingest_rem_mult = max(0.1, Chamber.max_damage - (Chamber.damage / Chamber.max_damage))

```

Comment:
This is going to override whatever is above--kinda defeating the purpose of species metabolic rates or what have you.

Also, is the intention to make it so that the `ingest_rem_mult` here ends up being `60`?